api/src/main/java/jakarta/data/expression/literal/ComparableLiteral.java

```diff
@@ -23,12 +23,40 @@
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
+import java.util.UUID;
 
 import jakarta.data.expression.ComparableExpression;
 
+/**
+ * <p>A {@linkplain Literal literal} for a sortable value, such as
+ * {@link Boolean}, {@link Character}, enumeration, and {@link UUID} values.
+ * </p>
+ *
+ * @param <T> entity type.
+ * @param <V> entity attribute type.
+ * @since 1.1
+ */
 public interface ComparableLiteral<T, V extends Comparable<?>>
         extends ComparableExpression<T, V>, Literal<T, V> {
 
+    /**
+     * <p>Creates a {@code ComparableLiteral} or subtype of
+     * {@code ComparableLiteral} that represents the given value.</p>
+     *
+     * <p>The most specific subtype of {@code ComparableLiteral}, such as
+     * {@link NumericLiteral#of(Number) NumericLiteral},
+     * {@link StringLiteral#of(String) StringLiteral}, or
+     * {@link TemporalLiteral#of(java.time.temporal.Temporal) TemporalLiteral},
+     * should be used instead wherever possible.</p>
```

Comment:
Is this better as an `@apiNote`?

It's almost like we need a `@jakartaDataImplementorNote`.

Reply:
> It's almost like we need a `@jakartaDataImplementorNote`.

I was actually intending this for the end user when they have a situation where they need to create a literal, encouraging them to use the most specific subtype so that they will get the most usefulness out of it.  I suppose that might already be obvious to them though if they had the use case for wanting the literal in the first place then they probably also know which type they want.

Reply:
Ah ok.